File: deploy/infrastructure/ssh_remote_exec.go
package infrastructure

import "github.com/koyeo/nest/execer"

// SSHRemoteExec implements domain.RemoteExec using SSH.
type SSHRemoteExec struct {
	server *execer.Server
}

// NewSSHRemoteExec creates a new SSHRemoteExec.
func NewSSHRemoteExec(server *execer.Server) *SSHRemoteExec {
	return &SSHRemoteExec{server: server}
}

func (r *SSHRemoteExec) Exec(command string) error {
	return r.server.CombinedExec(command)
}

func (r *SSHRemoteExec) ExecPipe(command string) error {
	return r.server.PipeExec(command)
}

File: apps/ship/src/deploy/infrastructure/ssh-remote-exec.ts
import type { RemoteExec } from "../domain/interfaces.js";
import type { SshServer } from "../../execer/ssh-server.js";

export class SshRemoteExec implements RemoteExec {
  constructor(private readonly server: SshServer) {}

  async exec(command: string): Promise<void> {
    await this.server.combinedExec(command);
  }

  execPipe(command: string): Promise<void> {
    return this.server.pipeExec(command);
  }
}
